app/plugin_api_test.go

```diff
@@ -1832,3 +1832,25 @@ func TestPluginAPIUpdateCommand(t *testing.T) {
 	require.Equal(t, team1.Id, newCmd4.TeamId)
 
 }
+
+func TestPluginNotificationWillBeSent(t *testing.T) {
```

Comment:
@hanzei , do you think we should have some other integration tests here? Perhaps testing a few of the notification types (at-user, at-channel, etc)?  
TBH, I'm not sure of the best way to do that from this test. Maybe you have some insight on how to do it?

Reply:
Does the test in `app/plugin_hooks_test.go` cover your request? (OT: Maybe we should fall back on only using ` app/plugin_api_tests`)

Reply:
@hanzei yes, that's good, thanks!

Up to you about using `app/plugin_api_tests`, I have no strong opinion